### ISSUE TITLE
Servercrash with SQL-Plugin from Dan

### ISSUE THREAD

**Issue by KingCodename** [Open-GTO/FCNPC#13]
We are using the SQL-Plugin from Dan (https://github.com/udan11/samp-plugin-sql). The server crash when there is a mistake in a quey which was executed. Please fix the bug.

Crash information:
```
=========================== FCNPC Exception Handler ============================
Signal 11 (Segmentation fault), address is (nil) from 0xf2b0ab98

Backtrace:
  #0 in plugins/fcnpc.so(_ZN17CExceptionHandler24ExceptionHandlerCallbackEiP7siginfoPv+0x164) [0xf2233c14]
  #1 in linux-gate.so.1(__kernel_rt_sigreturn+0) [0xf31fb410]
  #2 in plugins/sql.so(_ZN7Natives12sql_num_rowsEP6tagAMXPi+0x88) [0xf2b0ab98]
  #3 in /home/samp/sampServer() [0x80950e4]
  #4 in plugins/crashdetect.so(_ZN11CrashDetect13DoAmxCallbackEiPiS0_+0x49b) [0xf2ea594b]
  #5 in plugins/crashdetect.so(+0x1b8f8) [0xf2ea88f8]
  #6 in plugins/crashdetect.so(amx_Exec+0x18e6) [0xf2eac916]
  #7 in plugins/crashdetect.so(_ZN11CrashDetect9DoAmxExecEPii+0x96) [0xf2ea4be6]
  #8 in plugins/crashdetect.so(+0x1b659) [0xf2ea8659]
  #9 in plugins/streamer.so(+0x82ea) [0xf20882ea]

Information:
  System: GNU/Linux
  SA-MP: 0.3.7 R2-1
  FCNPC: 1.0.4

Functions:
  FUNC_CPlayerPool__DeletePlayer: 0x80d0a90
  FUNC_CPlayer__Kill: 0x80cb220
  FUNC_CPlayer__EnterVehicle: 0x80cc1c0
  FUNC_CPlayer__ExitVehicle: 0x80cc340
  FUNC_CPlayer__SpawnForWorld: 0x80ccfc0
  FUNC_GetVehicleModelInfo: 0x80d5d30
  FUNC_CConsole__GetIntVariable: 0x80a0070
  FUNC_ClientJoin_RPC: 0x80b0030

Pointers:
  VAR_ServerAuthentication: 0x81aa8a8
  VAR_NetVersion: 0xfd9

Offsets:
  OFFSET_RemoteSystemManager: 0x334
  OFFSET_RemoteSystemSize: 0xc69
  OFFSET_RemoteSystem__ConnectMode: 0xc62
  OFFSET_RemoteSystem__Unknown: 0xc67
=========================== FCNPC Exception Handler ============================
```

**Comment by ziggi** [comment]
I think what is not a FCNPC problem. FCNPC just intercept a segfault signal. Did you try it without FCNPC?
P.S. Why you using this very old plugin?

**Comment by KingCodename** [comment]
Yes I tried it without loading FCNPC.so and it works fine.

**Comment by KingCodename** [comment]
Here is another crash, but not in combination with a faulty query:

> =========================== FCNPC Exception Handler ============================
> Signal 11 (Segmentation fault), address is 0x101000b from 0x807cfe7
> 
> Backtrace:
>   #0 in plugins/fcnpc.so(_ZN17CExceptionHandler24ExceptionHandlerCallbackEiP7siginfoPv+0x164) [0xeb229c14]
>   #1 in linux-gate.so.1(__kernel_rt_sigreturn+0) [0xec54f410]
>   #2 in /home/samp/sampServer() [0x807cfe7]
>   #3 in /home/samp/sampServer() [0x807fc65]
>   #4 in /home/samp/sampServer() [0x808013d]
>   #5 in /home/samp/sampServer() [0x8074c3a]
>   #6 in /home/samp/sampServer() [0x80763fe]
>   #7 in /lib32/libpthread.so.0(+0x6efb) [0xec52aefb]
>   #8 in /lib32/libc.so.6(clone+0x5e) [0xec30cefe]
> 
> Information:
>   System: GNU/Linux
>   SA-MP: 0.3.7 R2-1
>   FCNPC: 1.0.4
> 
> Functions:
>   FUNC_CPlayerPool__DeletePlayer: 0x80d0a90
>   FUNC_CPlayer__Kill: 0x80cb220
>   FUNC_CPlayer__EnterVehicle: 0x80cc1c0
>   FUNC_CPlayer__ExitVehicle: 0x80cc340
>   FUNC_CPlayer__SpawnForWorld: 0x80ccfc0
>   FUNC_GetVehicleModelInfo: 0x80d5d30
>   FUNC_CConsole__GetIntVariable: 0x80a0070
>   FUNC_ClientJoin_RPC: 0x80b0030
> 
> Pointers:
>   VAR_ServerAuthentication: 0x81aa8a8
>   VAR_NetVersion: 0xfd9
> 
> Offsets:
>   OFFSET_RemoteSystemManager: 0x334
>   OFFSET_RemoteSystemSize: 0xc69
>   OFFSET_RemoteSystem__ConnectMode: 0xc62
>   OFFSET_RemoteSystem__Unknown: 0xc67
> =========================== FCNPC Exception Handler ============================

**Comment by ziggi** [comment]
Can you show me your FCNPC script?

**Comment by KingCodename** [comment]
I can show you the code auf your FCNPC functions:
```c
#include <fcnpc>

new npcID = INVALID_PLAYER_ID;

public OnFilterScriptInit()
{
	npcID = FCNPC_Create("npcName");
	FCNPC_Spawn(npcID, 1, 1873.6505, -2381.6863, 13.5547);
	return true;
}

public OnFilterScriptExit()
{
	if (npcID != INVALID_PLAYER_ID) {
		FCNPC_Destroy(npcID);
	}
	return true;
}

public FCNPC_OnSpawn(npcid)
{
	if (npcid == npcID) {
		FCNPC_StartPlayingPlayback(npcid, "recordfile");

		Some SetPlayerAttachedObject...
	}
	return true;
}
```

**Comment by ziggi** [comment]
Should be fixed in [v1.0.5](https://github.com/Open-GTO/FCNPC/releases/tag/v1.0.5)
